File: auth_setup.js
const { chromium } = require('playwright');
const fs = require('fs');
const readline = require('readline');

const authPath = 'auth.json';

// 터미널 입력을 기다리는 유틸리티
function waitEnter(msg) {
    const rl = readline.createInterface({ input: process.stdin, output: process.stdout });
    return new Promise(resolve => rl.question(msg, answer => { rl.close(); resolve(answer); }));
}

async function run() {
    console.log('[LOG] 프로그램 시작...');

    // 1. 기존 파일 확인 및 유효성 검사
    if (fs.existsSync(authPath)) {
        console.log(`[LOG] '${authPath}' 파일을 발견했습니다. 유효성을 검사합니다...`);
        try {
            const browser = await chromium.launch({ headless: true });
            console.log('[LOG] 검증용 임시 브라우저 실행 성공');
            
            const context = await browser.newContext({ storageState: authPath });
            const page = await context.newPage();
            
            console.log('[LOG] 쿠팡 접속 시도 중...');
            await page.goto('https://my.coupang.com/purchase/list', { waitUntil: 'domcontentloaded', timeout: 10000 });
            
            const currentUrl = page.url();
            console.log(`[LOG] 현재 URL: ${currentUrl}`);

            if (!currentUrl.includes('login.coupang.com')) {
                console.log('✅ [결과] 기존 세션이 유효합니다. 자동화를 시작합니다.');
                await browser.close();
                return;
            }
            console.log('[LOG] 세션이 만료되어 재인증이 필요합니다.');
            await browser.close();
        } catch (e) {
            console.log(`[ERROR] 유효성 검사 중 에러 발생: ${e.message}`);
        }
    } else {
        console.log(`[LOG] '${authPath}' 파일이 없습니다. 새로 생성을 시작합니다.`);
    }

    // 2. 사용자 안내
    console.log('\n' + '='.repeat(50));
    console.log('1. CMD창을 열고 아래 명령어를 입력하세요:');
    console.log('   netstat -ano | findstr :9222');
    console.log('2. 만약 데이터가 있다면 모든 크롬창을 꺼야합니다. (아래 명령어를 수행하면 모든 크롬창이 꺼집니다):');
    console.log('   taskkill /F /IM chrome.exe /T');
    console.log('2. 다음 명령어를 실행하세요:');
    console.log('   "C:\\Program Files\\Google\\Chrome\\Application\\chrome.exe" --remote-debugging-port=9222 --user-data-dir="C:\\temp\\chrome_debug" --no-first-run');
    console.log('3. 쿠팡 홈페이지에 로그인 한 후 아래 명령어를 수행하세요 (결과가 없으면 위의 과정을 재처리하세요):');
    console.log('   netstat -ano | findstr :9222');
    console.log('='.repeat(50));

    await waitEnter('\n👉 위 단계를 모두 마쳤다면 [엔터]를 눌러주세요...');

    // 3. CDP 연결 및 저장 (사용자 성공 코드 기반)
    try {
        console.log('[LOG] http://127.0.0.1:9222 에 연결을 시도합니다...');
        const browser = await chromium.connectOverCDP('http://127.0.0.1:9222');
        console.log('[LOG] 브라우저 연결 성공!');

        const contexts = browser.contexts();
        console.log(`[LOG] 활성화된 컨텍스트 수: ${contexts.length}`);
        
        const defaultContext = contexts[0];
        const pages = defaultContext.pages();
        console.log(`[LOG] 열려있는 페이지 수: ${pages.length}`);

        const page = pages[0] || await defaultContext.newPage();
        console.log(`[LOG] 대상 페이지 URL: ${page.url()}`);

        console.log('[LOG] 인증 정보를 추출하여 저장 중...');
        await defaultContext.storageState({ path: authPath });
        
        console.log(`✅ [결과] '${authPath}' 저장 완료!`);
        await browser.close();
        console.log('[LOG] 브라우저 연결을 해제하고 종료합니다.');
    } catch (error) {
        console.error('\n❌ [ERROR] 연결 최종 실패');
        console.error(`메시지: ${error.message}`);
        console.error('팁: 크롬이 9222 포트로 정말 띄워져 있는지 확인하세요.');
        process.exit(1);
    }
}

run();

File: auth_setup.py
import os
import json
from playwright.sync_api import sync_playwright
import os
import sys

# [핵심 추가] Playwright가 임시 폴더가 아닌 시스템 전역 브라우저 경로를 사용하게 함
if getattr(sys, 'frozen', False):
    # EXE로 실행 중일 때
    os.environ['PLAYWRIGHT_BROWSERS_PATH'] = '0'

AUTH_PATH = 'auth.json'

def check_login_status(page):
    try:
        # 주문 목록 페이지 접속 시도
        page.goto('https://my.coupang.com/purchase/list', wait_until='domcontentloaded', timeout=10000)
        # 로그인 페이지로 리다이렉트 되었는지 확인
        return 'login.coupang.com' not in page.url
    except:
        return False

def run_setup():
    with sync_playwright() as p:
        # 1. 기존 세션 유효성 확인
        if os.path.exists(AUTH_PATH):
            print(f"\n[LOG] '{AUTH_PATH}' 파일을 확인 중...")
            browser = p.chromium.launch(headless=True, channel="chrome")
            context = browser.new_context(storage_state=AUTH_PATH)
            page = context.new_page()
            
            if check_login_status(page):
                print("✅ 기존 세션이 유효합니다.")
                browser.close()
                return
            print("⚠️ 세션이 만료되었습니다.")
            browser.close()

        # 2. 안내 메시지
        print("\n" + "="*60)
        print("🔐 쿠팡 로그인 인증 설정")
        print("="*60)
        print("1. 모든 크롬 창을 닫으세요.")
        print("2. [윈도우+R] -> 아래 명령어 입력:")
        print("   chrome.exe --remote-debugging-port=9222 --user-data-dir=\"C:\\temp\\chrome_debug\"")
        print("3. 크롬에서 로그인 후 엔터를 누르세요.")
        print("="*60)
        
        input("\n👉 로그인을 완료했다면 [엔터]를 눌러주세요...")

        # 3. CDP 연결 및 저장
        try:
            print("\n[연결] 브라우저에 접속 중...")
            browser = p.chromium.connect_over_cdp("http://127.0.0.1:9222")
            default_context = browser.contexts[0]
            
            # 쿠팡 페이지로 이동하여 최종 확인
            page = default_context.pages[0] if default_context.pages else default_context.new_page()
            page.goto("https://www.coupang.com/")
            
            if '로그아웃' in page.content() or check_login_status(page):
                default_context.storage_state(path=AUTH_PATH)
                print(f"✅ 인증 정보 저장 완료! ({AUTH_PATH})")
            else:
                print("❌ 실패: 로그인이 감지되지 않았습니다.")
            
            browser.close()
        except Exception as e:
            print(f"\n❌ 연결 실패: {e}")

if __name__ == "__main__":
    run_setup()